### PR TITLE
MLP-154 - Integrate DI

### PR DESCRIPTION
**Pull request by lpelizzaMS**
# Background

We need to implement DI on our project. I have created two possible solutions one of them with Typhoon and the another with Dip. 
When I implemented the first one I found a limitation related to this -> "Every class you want to inject has to be a subclass of NSObject in some way (either by subclassing or adding @objc modifier)."
So, I chose the second option which doesn't have that limitation. 

# Changes done

I've implemented Dip which is a framework that solves the dependency injection paradigm. Also, I've included an example which uses the view controller and view model that already existed.

# Approvers required
@mvazquez-ms @mmaldini33  @jsmuguruza  @bsztamfater-ms

### CODE REVIEW

**Review by mmaldini33 — APPROVED** [review]

File: IOS/SwiftSeedProject/SwiftSeedProject/Interfaces/ViewControllers/LoginViewController.swift [hunk 1]
```diff
@@ -32,3 +33,5 @@ class LoginViewController: UIViewController {
     }
 
 }
+
+extension LoginViewController: StoryboardInstantiatable { }
```
Comment (mvazquez-ms) [inline IOS/SwiftSeedProject/SwiftSeedProject/Interfaces/ViewControllers/LoginViewController.swift:37]: What's the purpose of this? Is it required by Dip?

File: IOS/SwiftSeedProject/SwiftSeedProject/AppDelegate.swift [hunk 2]
```diff
@@ -7,12 +7,14 @@
...
 
 import UIKit
+import Dip
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
     var window: UIWindow?
-
+    private let container = DependencyContainer.configure()
```
Comment (mvazquez-ms) [inline IOS/SwiftSeedProject/SwiftSeedProject/AppDelegate.swift]: I think this line should be inside `didFinishLaunchingWithOptions `. The composition root could grow considerably and should only be built if the app was initialized successfully. With the current code, you have no control over that.

**Review by mvazquez-ms — APPROVED** [review]
Awesome simple PR!

File: IOS/SwiftSeedProject/SwiftSeedProject/Interfaces/ViewControllers/LoginViewController.swift (hunk 1 above)
Comment (lpelizzaMS) [reply IOS/SwiftSeedProject/SwiftSeedProject/Interfaces/ViewControllers/LoginViewController.swift:37]: This is required by Dip, it's needed for the instantiation of VC.

File: IOS/SwiftSeedProject/SwiftSeedProject/AppDelegate.swift (hunk 2 above)
Comment (lpelizzaMS) [reply IOS/SwiftSeedProject/SwiftSeedProject/AppDelegate.swift]: roger!

**Review by briansztamfater — APPROVED** [review]

**Pull request merged**